src/main/scala/cromwell/engine/package.scala

```diff
@@ -175,7 +175,7 @@ package object engine {
 
   case class SymbolStoreKey(scope: String, name: String, index: Option[Int], input: Boolean)
 
-  case class SymbolStoreEntry(key: SymbolStoreKey, wdlType: WdlType, wdlValue: Option[WdlValue]) {
+  case class SymbolStoreEntry(key: SymbolStoreKey, wdlType: WdlType, wdlValue: Option[WdlValue], isWorkflowOutput: Boolean) {
```

Comment:
I believe this is all changing so it's academic anyways, but when you have a boolean flag in your product type it's saying that you also have a sum type in there.

So what is being stated here is two things:
- The SymbolStoreEntry type has a key, a type and maybe a value
- The SymboleStoreEntry type is either a workflow output or not

So the structure should be something like this (I'm not going to bother thinking of good names, so just as an example):

    sealed trait SymbolStoreEntry {
      def key: SymbolStoreKey
      def wdlType: WdlType
      def wdlValue: Option[WdlValue]
    }
    final case class WorkflowOutputEntry(key: SymbolStoreKey, wdlType: WdlType, wdlValue: Option[WdlValue]) extends SymbolStoreEntry
    final case class SomeOtherEntry(key: SymbolStoreKey, wdlType: WdlType, wdlValue: Option[WdlValue]) extends SymbolStoreEntry

Now you have codified that difference in the type system.